src/DeltaLake/Kernel/Arrow/Converters/ArrowFfiSchemaConverter.cs

```diff
@@ -0,0 +1,72 @@
+// -----------------------------------------------------------------------------
+// <summary>
+// Converts FFI Arrow Schema into CArrowSchema.
+// </summary>
+//
+// <copyright company="The Delta Lake Project Authors">
+// Copyright (2024) The Delta Lake Project Authors.  All rights reserved.
+// Licensed under the Apache license. See LICENSE file in the project root for full license information.
+// </copyright>
+// -----------------------------------------------------------------------------
+
+using Apache.Arrow;
+using Apache.Arrow.C;
+using DeltaLake.Kernel.Interop;
+
+namespace DeltaLake.Kernel.Arrow.Converters
+{
+    /// <summary>
+    /// Static methods to convert <see cref="FFI_ArrowSchema"/> schema to <see
+    /// cref="CArrowSchema"/>.
+    /// </summary>
+    /// <remarks>
+    /// <see cref="CArrowSchema"/> release property is internal for some reason,
+    /// but that's not the case for <see cref="FFI_ArrowSchema"/>. To do the
+    /// conversion, we need to initialize it with a known type and then
+    /// overwrite it.
+    ///
+    /// TOOD: Need to check with the Arrow C# authors why they do stuff like
+    /// this that makes life hard, i.e. why is release the only property that's
+    /// internal, without a setter?
```

Comment:
In normal usage, letting this be set would be a footgun. That said, I don't know why you'd need this; the whole point of the C API is that it's blittable and you can just do
```
        private static unsafe Apache.Arrow.Schema ConvertFFISchemaToArrowSchema(FFI_ArrowSchema* ffiSchema)
        {
            return CArrowSchemaImporter.ImportSchema((CArrowSchema*)ffiSchema);
        }
```

Reply:
Haha thanks for checking this @CurtHagenlocher :) - and for reaching out aside.

The Delta Kernel FFI interface exposes a `FFI_ArrowSchema` pointer, but that not the actual C interface, so `CArrowSchemaImporter.ImportSchema` doesn't work against it - please see line 477:

https://github.com/delta-incubator/delta-dotnet/blob/8f448dc26de2770df33c45fda044550c991e6c0f/src/DeltaLake/Kernel/Interop/Interop.cs#L477

`FFI_ArrowSchema` looks _almost_ similar to `CArrowSchema`, but it's not quite the same.

So I'm stuck copying pointers from the Delta Kernel FFI, and trying to feed it into the `CArrowSchemaImporter`, and that private `release` field gave me a lot of pain (hence the passive-aggressive docstring :)) - but the workaround you see in this file worked.

Reply:
So say if I try to cast as you suggested, I get:

![image](https://github.com/user-attachments/assets/b8910c01-1bb8-469d-9172-d85b1f370c86)

Reply:
Look more closely at the method prototype :P. `ImportSchema` returns an `ArrowSchema`, not a `CArrowSchema*`.

The definitions had better be 100% compatible as there is only one standard definition for the Arrow (C) FFI.

Reply:
Oh wow. I spent an hour trying to hack against this when a simple pointer cast would have done it 🥲

Thank you so much for reading through this section. I'm going to get rid of this `ArrowFfiSchemaConverter` completely if the tests pass with this method.

Keeping the comment open until this change is done.

Reply:
Yup, worked beautifully:

![image](https://github.com/user-attachments/assets/3814cdc8-bdd6-4600-a00a-91c56339a0d8)

Remove this whole file. Thanks again!